providers/keycloak.go

```diff
@@ -95,5 +128,29 @@ func (p *KeycloakProvider) GetEmailAddress(ctx context.Context, s *sessions.Sess
 		}
 	}
 
+	if len(p.Roles) > 0 {
```

Comment:
Can you refactor this into another method that does the role authorization? This addition makes this `GetEmailAddress` method too complex: https://codeclimate.com/github/oauth2-proxy/oauth2-proxy/pull/767

Reply:
Sure thing, but I believe after refactoring block into its own method it would still exceed the number of allowed returns (it would become 1 less, leaving it at 5, which still exceeds the 4 allowed) so that would not fix the issue. Would you still suggest to refactor it?

Reply:
Code climate is advisory, take it with a pinch of salt. Would be good still to have a dedicated role check function so basically just extract the contents of this if to a function I would say

Reply:
Got it, I've extracted the logic into a separate function.